fedimint-api-client/src/api/global_api.rs

```diff
@@ -129,12 +132,56 @@ where
         .map_err(|e| anyhow!(e.to_string()))
     }
 
+    fn select_peers_for_status(&self) -> impl Iterator<Item = PeerId> + '_ + Clone {
+        // TODO: better selection
+        self.all_peers().iter().cloned()
+    }
+
     async fn get_session_status_raw(
         &self,
         block_index: u64,
         decoders: &ModuleDecoderRegistry,
     ) -> anyhow::Result<SessionStatus> {
         debug!(target: LOG_CLIENT_NET_API, block_index, "Fetching block's outcome from Federation");
+        let verify = |_: &SignedSessionOutcome| -> bool { todo!() };
+        // TODO: check api version
```

Comment:
Probably just make it a new function, and let the caller verify. I think all over the code we do the api versioning checks in the higher layer. Though passing api version here wouldn't be bad either.